packages/iTwinUI-react/src/core/Toast/Toaster.tsx

```diff
@@ -115,8 +107,37 @@ export default class Toaster {
     this.updateView();
   }
 
+  private async createContainer() {
+    const container = getContainer(TOASTS_CONTAINER_ID) ?? getDocument()?.body;
+    if (!container) {
+      return;
+    }
+
+    const toastWrapper = <ToastWrapper ref={this.toastsRef} />;
+
+    try {
+      if (Number(React.version.split('.')[0]) > 17) {
```

Comment:
Have you checked whether this is working in prod builds in 17 and 18?

Reply:
I have checked only dev builds

Reply:
ok, great catch!
Does not work neither in vite, nor in cra apps.. :(

Reply:
will `try`/`catch` work instead? like if you `await import` from `react-dom/client` on react 17, does it throw an error?

Reply:
It fails on build time...doesnt matters this code..

Reply:
What's the error? I think there's a way to hack around it at build time.

Reply:
![image](https://user-images.githubusercontent.com/81580355/169044488-68b96e20-43a8-45c3-a072-eaae360e0d01.png)

Reply:
found out last week dynamic imports are resolved at build time. with webpack you can ignore it with magic comments, not sure about vite

Reply:
Rollup also has additional config option, which might help, but it is not a solution for us, imo..
We would break current setup of the apps, so that technically should come with a major release.

Reply:
looks like esbuild is the only one that handles this in a sensible way
https://github.com/evanw/esbuild/commit/6be8e44beba8e765c2b99683608540ce40d6a58f

Reply:
Ok, I have tested the solution we have in:
- Rollup build with Vite app
- Webpacke build CRA app
- next.js app

I have tried both dev and prod builds and stuff seemed to work fine. Also no warnings were displayed in v18 (that old render is being used).
If someone is still up to, I would appreciate if you could run one more time through Vite or CRA app (separate one, not the playground we have, just to be sure), so that my local builds were not messed up..
But it looks for now we can go with this change and hopefully we will refactor fully toasts to get rid of render in future major releases, before react decided to get rid of their barrel export of `createRoot`.

Thank you @mayank99 for the hacky inspiration, but I think looks much better than antd one 😄

Reply:
I tested on CRA that has v17 and on Vite with v18 and no errors in dev or in prod builds.